.github/workflows/android-device-test/download-benchmark-results.mjs

```diff
@@ -0,0 +1,113 @@
+// This script parses logcat output from AWS Device Farm to extract benchmark results.
+// It also uploads the results to S3.
+// android-benchmark-render/{gitRevision}/{deviceModel}.json
+
+import {
+  DeviceFarmClient,
+  ListJobsCommand,
+  ListArtifactsCommand,
+} from "@aws-sdk/client-device-farm";
+import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
+
+import https from "node:https";
+import readline from "node:readline";
+
+const deviceFarmClient = new DeviceFarmClient({ region: "us-west-2" });
+
+function usage() {
+  console.error("Downloads benchmark results from AWS Device Farm");
+  console.error(`Usage: node ${process.argv.at(1)} RUN_ARN`);
+  process.exit(1);
+}
+
+if (process.argv.length !== 3) usage();
+
+const arn = process.argv.at(2);
+
+const listJobsCommand = new ListJobsCommand({ arn });
+const { jobs } = await deviceFarmClient.send(listJobsCommand);
+
+const passedJobs = jobs.filter((job) => job.result === "PASSED" && job.arn);
+const failedJobs = jobs.filter((job) => job.result !== "PASSED" || !job.arn);
+
+failedJobs.forEach((job) => {
+  console.error(
```

Comment:
Should this fail the build? Otherwise I'm not sure how helpful this console error is in a CI environment...

Reply:
Nope. Sometimes benchmarks fail and it's nice to see in the logs which did.